script.js

```diff
@@ -1,4 +1,103 @@
+var myOutputValue = "";
+var userWin = 0;
+var computerWin = 0;
+var totalPlay = 0;
+var currentGameMode = "Waiting for username";
+var userName = "";
+
+// Computer generates random Number
+var randomNumber = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomInteger = Math.floor(randomDecimal);
+  return randomInteger;
+};
+
+//Computer assigns number to "scissors", "paper", or "stone"
+var computerGuesses = function () {
+  var computerGuess = "";
+  var guess = randomNumber();
+  if (guess == 0) {
+    computerGuess = "scissors";
+  }
+  if (guess == 1) {
+    computerGuess = "paper";
+  }
+  if (guess == 2) {
+    computerGuess = "stone";
+  }
+  return computerGuess;
+};
+
+// User input "scissors", "paper", or "stone";
+// if user input beats computer's, user win
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var userInput = input.toLowerCase();
+  //Waiting for user input
+  if (currentGameMode == "Waiting for username") {
+    //if the input's value for not a number is false, the user has to type again
+    if (Number.isNaN(Number(input)) == false) {
+      myOutputValue = "Sorry please enter a Name.";
+    } else {
+      userName = input;
+      currentGameMode = "SPS game";
+      myOutputValue =
+        "Hello " + userName + "<br><br> Please key in your guess.";
+    }
+  }
+
+  //When user input and the SPS game mode is switched on
+  else if (currentGameMode == "SPS game") {
+    var computersGuess = computerGuesses();
+    var commonOutput =
+      "The Computer chose " +
+      computersGuess +
+      "<br><br> You chose " +
+      userInput;
+    //if we beat the computer, we win
+    if (
+      (computersGuess == "paper" && userInput == "scissors") ||
+      (computersGuess == "stone" && userInput == "paper") ||
+      (computersGuess == "scissors" && userInput == "stone")
+    ) {
+      userWin = userWin + 1;
+      totalPlay = totalPlay + 1;
+      myOutputValue =
+        commonOutput +
+        ".<br><br> You won! <br><br> Now you can type scissors, paper or stone to play another round! <br><br> So far " +
+        userName +
+        ", you've been winning: " +
+        userWin +
+        "/" +
+        totalPlay +
+        " turns. Pretty good!";
```

Comment:
Good job combining all the variables into one myOutputValue.

Some ways to improve on this are to:
1. add ''''<br><br> Now you can type scissors, paper or stone to play another round!<br><br> So far " + userName + ", you've been winning: " + userWin + "/" +totalPlay +" turns. Pretty good!";''''' 
- to your my output value since it repeats. 
- Instead you can consider creating a new variable called var result and equate that to <br><br> It's a draw! / <br><br> You lose! Bummer. /  <br><br> You won! 
- since these are the only parts that differ in each condition. That way you do not have repeating code.

2. Use javastring literals `` instead of ' ' so that you can combine your string values with your variables in one sentence without repeated plus signs. This will save you a lot of time.
https://www.w3schools.com/js/js_string_templates.asp